app/lib/actions/pulp3/repository/save_artifact.rb

```diff
@@ -0,0 +1,21 @@
+module Actions
+  module Pulp3
+    module Repository
+      class SaveArtifact < Pulp3::Abstract
+        def plan(file, repository, tasks, unit_type_id)
+          file_name = file[:filename]
+          plan_self(:file_name => file_name, :repository_id => repository.id, :tasks => tasks, :unit_type_id => unit_type_id)
+        end
+
+        def run
+          artifact_href = input[:tasks].last[:created_resources].first
+          content_type = input[:unit_type_id]
+          content_backend_service = SmartProxy.pulp_master.content_service(content_type)
+          data = content_backend_service.content_class.new(relative_path: input[:file_name], _artifact: artifact_href)
+          content_unit = content_backend_service.content_api.create(data)
```

Comment:
Is this relative_path a file specific thing?  I'm trying to understand why it would be needed when creating a content unit generically

Reply:
You are right..It is content specific..I see this on files and docker but not on collections..I will change this..